Sales Order Agent doc update (#444)
microsoft/Dynamics-365-FastTrack-Implementation-Assets: AI ERP Agents/Sales Order Agent/readme.md

# Sales Order Processor Agent

# 🧩 Use Case

> Streamlining and automating the intake of **sales orders** received as email attachments for seamless integration into Microsoft Dynamics 365 Finance and Operations.

Reviewer: ```suggestion
Streamlining and automating the intake of *sales orders* received as email attachments for seamless integration into Microsoft Dynamics 365.
```

There is no brand called "Microsoft Dynamics 365 Finance and Operations".

# 🛠️ Approach

> The solution will lead with our **first-party agent Document Processor for document extraction, and content validation**. Once the content is validated, we will use a **second agent to create the order and order lines into Dynamics 365 Finance and Operations.**

Reviewer: ```suggestion
The solution will lead with the built-in agent **Document Processor** for document extraction and content validation. Once the content is validated, we use a second agent to create the order and order lines into Dynamics 365.
```

## 📄 Document Processor Agent
**Document Processor Agent** is a robust managed agent, packaged solution for end-to-end document processing including **extraction, validation and human monitoring**. It does not require training custom models, instead a relevant sample document can be uploaded, and the maker can configure the attributes that should be extracted and if any validation rules to be applied. For more details please see: https://learn.microsoft.com/en-us/microsoft-copilot-studio/template-managed-document-processor.
### Configuration Wizard
When configuring the Document Processor Agent, to achieve a more deterministic JSON schema that you can leverage in the Sales Order agent to parse the data for the downstream systems, it is recommended to update the Document Processor extraction rules and include a similar prompt: 
![docProcessorValidationRulesCustom](images/docProcessorValidationRulesCustom.png)
![docProcessorValidationCustomAdvanced](images/docProcessorValidationCustomAdvanced.png)

## 🤖 Sales Order Processor Agent

> **Sales Order Processor Agent** is an agent template to help support **an end-to-end document processing flow into downstream apps such as Dynamics 365 Finance and Operations**. The agent includes sales order input validations for customer name and product codes, creation of a sales order header and related lines, as well as acknowledgement emails.

Reviewer: ```suggestion
**Sales Order Processor Agent** is an agent template to help support an end-to-end document processing flow into downstream apps such as Dynamics 365 Finance or Supply Chain Management. The agent includes sales order input validations for customer name and product codes, creation of a sales order header and related lines, as well as acknowledgement emails.
```
